llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -17910,6 +17910,13 @@ AArch64TargetLowering::isDesirableToCommuteWithShift(const SDNode *N,
   SDValue ShiftLHS = N->getOperand(0);
   EVT VT = N->getValueType(0);
 
+  if (!ShiftLHS->hasOneUse())
```

Comment:
Does this logic need to be added to the default implementation of `isDesirableToCommuteWithShift` for targets that don't override it? I believe the default returns `true`

Reply:
Some test cases will fail~Brief look, it seems that there is no benefit
![image](https://github.com/user-attachments/assets/85ac5b76-5b2f-4d2b-899a-384ba3f35cba)

Reply:
AArch64 overrrides the hook. I wasn't asking you to change what you already did for AArch64. So why do AArch64 tests change?

I was asking you to additionally change the default implementation.

Reply:
Any problems with this Patch？

Reply:
There was a holiday in the US on Thursday so I'm a little behind on reviews right now.